### ISSUE TITLE
Initialization of hyperparameter search space fails when the first preprocessor cannot be used with given estimators

### ISSUE THREAD

**Issue by teresaconc** [automl/auto-sklearn#631]
Initialization of hyperparameter search space fails when the first preprocessor method can not be used with none of the estimators.

Example:
Using in the fit method:
```
 include_preprocessors = ['kernel_pca','select_rates']
 include_estimators = ['decision_tree','random_forest']
```
an error is raised: **_"Cannot find a legal default configuration."_** in [ _get_hyperparameter_search_space: line 216](https://github.com/automl/auto-sklearn/blob/master/autosklearn/pipeline/classification.py#L216). This raises an error because kernel_pca can not be used with neither decision_tree nor random_forest.

However, if I reverse the include_preprocessors list:
` _include_preprocessors_ = ['select_rates', 'kernel_pca']`
The program does not exit and the fit method is successful. It does not uses 'kernel_pca' but it does search for models with 'select_rates'

**Comment by mfeurer** [comment]
Indeed, this is an issue. Ideally, one would check every combination of preprocessor and classifier and then choose a legal one if possible. Would you like to work on a fix for this?

**Comment by teresaconc** [comment]
I haven't had time to look into this. But will hopefully do it soon

**Comment by mfeurer** [comment]
That would be great!
